boskos/janitor/janitor.py

```diff
@@ -148,13 +149,17 @@ def clear_resources(project, cols, resource):
             else:
                 base.append('--global')
 
-        print 'Call %r' % base
-        try:
-            subprocess.check_call(base + list(items))
-        except subprocess.CalledProcessError as exc:
-            if not resource.tolerate:
-                err = 1
-            print >>sys.stderr, 'Error try to delete resources: %r' % exc
+        print 'going to delete %d %s' % (len(items), resource.name)
+        # try to delete at most $rate_limit items at a time
+        for idx in xrange(0, len(items), rate_limit):
+            clean = items[idx:idx+rate_limit]
```

Comment:
`idx+rate_limit` is not necessarily smaller than `len(items)`

Reply:
But python doesn't care about index out of range on a slice.... WTF

Reply:
python is great :-)